yjit/src/codegen.rs

```diff
@@ -1429,70 +1431,70 @@ gen_newrange(jitstate_t *jit, ctx_t *ctx, codeblock_t *cb)
 
     return YJIT_KEEP_COMPILING;
 }
+*/
 
-static void
-guard_object_is_heap(codeblock_t *cb, x86opnd_t object_opnd, ctx_t *ctx, uint8_t *side_exit)
+fn guard_object_is_heap(cb: &mut CodeBlock, object_opnd: X86Opnd, ctx: &mut Context, side_exit: CodePtr)
 {
     add_comment(cb, "guard object is heap");
 
     // Test that the object is not an immediate
-    test(cb, object_opnd, imm_opnd(RUBY_IMMEDIATE_MASK));
+    test(cb, object_opnd, imm_opnd(RUBY_IMMEDIATE_MASK as i64));
```

Comment:
For masks like that you should probably use `uimm_opnd()` (unsigned) and cast as u64.